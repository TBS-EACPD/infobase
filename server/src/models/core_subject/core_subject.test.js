const _ = require("lodash");


const crso_query  = `
query CRSOTestQuery ($lang: String!){
  root(lang: $lang){
    org(org_id:"128"){
      acronym
      crsos {
        id
      }
    }
    crso(id:"HRSD-SO0491"){
      name
      programs {
        id
      }
      org {
        id
      }
    }
    program(id:"HRSD-AMQ00"){
      crso {
        id
      }
    }
  }
}
`;


const { execQuery } = global;

describe("subject linkage", function(){

  it("crso root, up and down linkage", async ()=> {
    const data = await execQuery(crso_query, {});
    return expect(data).toMatchSnapshot();
  });

})

const org_fields_query = `
query IgocTestQuery ($lang: String!){
  root(lang: $lang){
    
    org(org_id:"128"){
      inst_form {
        id
      }
      article_1
      article_2
      enabling_instrument
      auditor
      notes
      faa_schedule_institutional
      incorp_yr
      
      
      ministry {
        name
        id
        orgs {
          id
        }
      }
      ministers {
        id
        name
      }
    }
    org_with_notes: org(org_id: "10"){
      notes
    }
    org_with_auditor: org(org_id: "97"){
      auditor
    }
    org_with_end_date: org(org_id:"291"){
      end_yr
    }
    org_with_faa_hr: org(org_id:"296"){
      faa_schedule_hr_status
    }
    org_with_fed_ownership: org(org_id: "285"){
      fed_ownership
    }
  }
}
`

describe("igoc linkage and fields", function(){

  it("igoc linkage and fields", async ()=> {
    const data = await execQuery(org_fields_query, {});
    return expect(data).toMatchSnapshot();
  });

});

const program_query = `
query ProgramTestQuery ($lang: String!){
  root(lang: $lang){
    program(id: "AGR-AAA00"){
      id
      activity_code
      is_internal_service
      description
      
    }
    internal_service_program: program(id:"AGR-ISC00"){
      is_internal_service
    }
    inactive_program: program(id: "TBC-DAC00"){
      is_active
    }
    
  }
}
`;

describe("program basic fields", function(){

  it("program basic fields", async ()=> {
    const data = await execQuery(program_query, {});
    return expect(data).toMatchSnapshot();
  });

});
