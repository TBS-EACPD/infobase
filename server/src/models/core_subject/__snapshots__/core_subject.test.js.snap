// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`igoc linkage and fields igoc linkage and fields 1`] = `
Object {
  "data": Object {
    "root": Object {
      "org": Object {
        "article_1": null,
        "article_2": "de",
        "auditor": null,
        "enabling_instrument": "National Defence Act, R.S.C. 1985, c. N-5",
        "faa_schedule_institutional": null,
        "incorp_yr": null,
        "inst_form": null,
        "ministers": null,
        "ministry": null,
        "notes": null,
      },
      "org_with_auditor": Object {
        "auditor": "Ernst & Young LLP and PricewaterhouseCoopers LLP",
      },
      "org_with_end_date": Object {
        "end_yr": null,
      },
      "org_with_faa_hr": Object {
        "faa_schedule_hr_status": null,
      },
      "org_with_fed_ownership": Object {
        "fed_ownership": null,
      },
      "org_with_notes": Object {
        "notes": null,
      },
    },
  },
}
`;

exports[`program basic fields program basic fields 1`] = `
Object {
  "data": Object {
    "root": Object {
      "inactive_program": Object {
        "is_active": false,
      },
      "internal_service_program": Object {
        "is_internal_service": false,
      },
      "program": Object {
        "activity_code": "ATT00",
        "description": null,
        "id": "ND-ATT00",
        "is_internal_service": false,
      },
    },
  },
}
`;

exports[`subject linkage crso root, up and down linkage 1`] = `
Object {
  "data": Object {
    "root": Object {
      "crso": Object {
        "name": "Defence operations improve peace, stability and security wherever deployed",
        "org": Object {
          "id": "133",
        },
        "programs": Array [
          Object {
            "id": "ND-AUW00",
          },
          Object {
            "id": "ND-AUX00",
          },
          Object {
            "id": "ND-AUY00",
          },
          Object {
            "id": "ND-AUZ00",
          },
        ],
      },
      "org": Object {
        "acronym": null,
        "crsos": Array [
          Object {
            "id": "ND-BUN00",
          },
          Object {
            "id": "ND-BUO00",
          },
          Object {
            "id": "ND-BUP00",
          },
          Object {
            "id": "ND-BUQ00",
          },
          Object {
            "id": "ND-BUR00",
          },
          Object {
            "id": "ND-BUS00",
          },
          Object {
            "id": "ND-ISS00",
          },
          Object {
            "id": "ND-SO00224",
          },
          Object {
            "id": "ND-SO00225",
          },
          Object {
            "id": "ND-SO00226",
          },
          Object {
            "id": "ND-SO00227",
          },
          Object {
            "id": "ND-SO00228",
          },
          Object {
            "id": "ND-SO00229",
          },
          Object {
            "id": "ND-SO00230",
          },
        ],
      },
      "program": null,
    },
  },
}
`;
